### PR TITLE
Update OpenSSL version for libevent

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **libevent/2.1.12**

libevent is using OpenSSL 1.1.1j which is not the latest version available.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libevent/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 cmake_minimum_required(VERSION 3.1)
-project(test_package)
+project(test_package CXX)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
```
Comment (madebr) [inline recipes/libevent/all/test_package/CMakeLists.txt:5]: Looking at `test_package.cpp`, it is actually c.
So you can rename it to `test_package.c`.

Comment (uilianries) [reply recipes/libevent/all/test_package/CMakeLists.txt:5]: good catch

Comment (uilianries) [reply recipes/libevent/all/test_package/CMakeLists.txt:5]: done

**Review by madebr — APPROVED** [review]

File: recipes/libevent/all/conanfile.py [hunk 2]
```diff
@@ -49,14 +49,10 @@ def configure(self):
 
     def requirements(self):
         if self.options.with_openssl:
-            self.requires("openssl/1.1.1j")
+            self.requires("openssl/1.1.1l")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        # temporary fix due to missing files in dist package for 2.1.11, see upstream bug 863
```
Comment (Croydon) [inline recipes/libevent/all/conanfile.py:56]: What made this workaround obsolete?

Comment (uilianries) [reply recipes/libevent/all/conanfile.py:56]: strip_root, it takes the extracted folder and rename to destination

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libevent/all/conanfile.py (hunk 2 above)
Comment (Croydon) [reply recipes/libevent/all/conanfile.py:56]: Yes, that helps when the directory names are different, but I don't really understand this inline comment. It is talking about missing files...?

Comment (uilianries) [reply recipes/libevent/all/conanfile.py:56]: Sorry, I though was about strip_root. About missing file is still the same problem, the fix is downloading that specific tar file. If you visit https://github.com/libevent/libevent/releases/tag/release-2.1.12-stable, you will see source code tar file, which is the regular way for CCI, but also there is another tar file named with "release"

**Review by SSE4 — APPROVED** [review]

**Pull request merged**